matrix-sdk/src/main/java/org/matrix/androidsdk/db/MXMediasCache.java

```diff
@@ -599,12 +599,6 @@ public String saveBitmap(Bitmap bitmap, String defaultFileName) {
             File file = new File(getFolderFile(null), filename);
             FileOutputStream fos = new FileOutputStream(file.getPath());
 
-            // We got an java.lang.IllegalStateException: Can't compress a recycled bitmap
-            if (bitmap.isRecycled()) {
-                Log.w(LOG_TAG, "Trying to compress a recycled Bitmap. Create a copy first.");
-                bitmap = Bitmap.createBitmap(bitmap);
-            }
```

Comment:
Are you sure it is necessary to remove this block? I've added it because on some device it fixes an issue (not observed on all devices).

Reply:
As far as I know the root cause of that issue was the other block I had removed below which I have proposed a fix for as well.